### DISCUSSION TITLE
Loop for on Teensy4. too much slow!!

### DISCUSSION THREAD

**Discussion by LeandroFernandes88** [micropython/micropython#15141] [m.iMX RT / Teensy 4.x]
hello everybody.
I'm making code to read around 4000 numpy array files from an sdcard and sorting the data.
In this process, I read the name of the files and store them in a list and start reading them 1 by 1 in a for loop structure.
The first 100 readings are very fast, around reading 500, the code starts to slow down, to the point where it becomes extremely slow after 1000 readings.
I got to the point of waiting 2 hours and 30 minutes for something that I thought I would get between 5 and 10 minutes of audio processing.
Could it be something placed in the hardware by PJRC to optimize this loop structure that could be presenting this delay in Micropython?

**Comment by andrewleech** [comment]
Can you share any of the code you use here? These kind of speed questions can be tricky to answer from high level description as there's lots of different things can can cause a slowdown. However they're usually to do with lists/buffers being expanded during use / in the loop.

**Reply by LeandroFernandes88** [reply]
thanks for the help [andrewleech](https://github.com/andrewleech)
I'll try to manage memory better, delete variables at the end of each loop and use gc, I will see if that solves the problem,
I'll make the code a little better and share it here.
